Kalela/BrightEvents
# Ft-api-authorization-154109633

### What does this PR do?
Creates functionality for token based authentication.
### How should this be manually tested?
Using postman, Run the following endpoints with appropriate inputs. Ensure to generate your token with the Login route, and pass this token in the header of each consequent endpoint(Except the register endpoint) with key-value 'x-token-access'.

POST /api/v2/auth/register    |Registers a user
POST /api/v2/auth/login    |Logs a registered user in
POST /api/v2/auth/logout    |Logs a logged in user out
POST /api/v2/auth/reset-password    |Resets a logged in users password
GET /api/v2/events    |Views all events
POST /api/v2/events    |Adds an event 
PUT /api/v2/events/<eventid>    |Edits an existing event
DELETE /api/v2/events/<eventid>    |Deletes an existing event
POST /api/v2/events/<eventid>/rsvp    |Sends an rsvp to a event
### What are the relevant pivotal tracker stories?
Setup token based authentication
ID: #154109633
